src/System.Linq.Parallel/src/System/Linq/Parallel/Channels/AsynchronousChannel.cs

```diff
@@ -596,14 +602,14 @@ private bool TryDequeueChunk(ref T[] chunk, ref bool isDone)
         //     The caller has verified that a chunk is available, i.e. the queue is non-empty.
         //
 
-        private T[] InternalDequeueChunk()
+        private T[]? InternalDequeueChunk()
         {
             Debug.Assert(!IsChunkBufferEmpty);
```

Comment:
💭 It seems like this assertion would mean this method cannot return null. Are we sure the return value annotation is correct?

Reply:
Seems like that was the intent, but it was not so cleat that is guaranteed, as i cannot guarantee that marked is nullable, but if the owner or whoever familiar with this code would say that assumption is guaranteed i would update accordingly